client/src/components/Portal/index.jsx

```diff
@@ -0,0 +1,21 @@
+import { useEffect } from 'react';
+import ReactDOM from 'react-dom';
+import PropTypes from 'prop-types';
+
+const Portal = ({ children }) => {
+    const portalRoot = document.getElementById('portal');
+    const element = document.createElement('div');
+
+    useEffect(() => {
+        portalRoot.appendChild(element);
+        return () => portalRoot.removeChild(element);
+    }, [portalRoot, element]);
+
+    return ReactDOM.createPortal(children, element);
+};
```

Comment:
co tu sie dzieje? :eyes:

Reply:
Utworzenie elementu Portal za pomocą hooks: useState i useEffect. Do elementu o id 'portal' dodaje element 'div', a po odmontowaniu jest usuwany. Portal u nas w projekcie jest wykorzystywany do modal'a. Chciałam zobaczyć jak to działa. I działa.
Umożliwia to przeniesienie struktury od rodzica, a zarazem nadal zachowuje się jak jego komponent - dziecko.
Wiem, niepotrzebnie zrobiony. Nawet nie zakładaliśmy, że będzie jakiś modal 😄

Reply:
hmmm... dodawanie divow w ten sposob jest troche zaprzeczeniem stosowania flow reacta i troche sie zastanawiam czy mozna to zrobic w standardowy sposob
:eyes:

Reply:
![image](https://user-images.githubusercontent.com/15021131/66269024-b6de8500-e843-11e9-8b02-3c1beffed5c3.png)

Reply:
Jeszcze widziałam z referencjami (`useRef()`), ale to też tworzony był `div`, aby go wykorzystać do `appendChild`.